src/shogun/solver/LDASolver.h

```diff
@@ -110,32 +111,31 @@ namespace shogun
 	void LDASolver<T>::compute_means()
 	{
 		index_t num_class = m_labels->get_num_classes();
-		auto data = m_features->get_feature_matrix();
```

Comment:
We should benchmark memory use xvalidation before and after this PR.... or maybe at a later point at least! :)

Reply:
sure! i would imagine this is only faster when the matrix being used is larger than the L3 cache... i wonder if there is a noticeable impact on smaller examples. there might also be some unwanted copies, i need to check the code a few mote times...

Reply:
i did some micro benchmarking with google benchmark and the iterator without subset is as fast as using the getter in the loop. However, so far the subset is always slower. The fact that we have to do two argsort operations (in begin and end) cannot really be compensated for..